### PR TITLE
Supports image textures with transparency.

### LINKED ISSUE #13: Texture mapping capability
@maxxfrazer 
**Question**:
Thanks for bug fix with the color change when creating FocusEntity. The second part of my question was about how to apply a different shape for the FocusEntity (by default it is a square, I need a circle (see [Ikea Place app](https://apps.apple.com/us/app/ikea-place/id1279244498))), besides how I should be able to add texture, not just color.

**Decision**:
You need to add this functionality as a must have.

**The alternatives I have considered**:
1. Using FocusEntity, I create it custom - I can apply MeshResource and colored, but there is no way to apply a texture:
```swift
let meshResource = MeshResource.generatePlane (width: 0.2, depth: 0.2)
let focusEntity = FocusEntity (on: self, focus: FocusEntityComponent (style: .colored (onColor: .red, offColor: .green, nonTrackingColor: .blue, mesh: meshResource)))
```
2. Without using FocusEntity, I created a ModelEntity with a solid fill or texture, after which I generated a rayCast from center of screen and positioned (moved) ModelEntity to a point on the detected plane:
```swift
// ...
let meshResource = MeshResource.generatePlane (width: width, depth: width, cornerRadius: 2)
var material: UnlitMaterial!
 if let url = Bundle.main.url (forResource: "Files/targetShapeTexture_001", withExtension: "png") {
 texture = try TextureResource.load (contentsOf: url)
material = UnlitMaterial()
material! .baseColor = .texture (texture!)
material! .tintColor = UIColor (red: 25, green: 25, blue: 25, alpha: 0.5)
// ...
// modelEntity
let modelEntity = ModelEntity (mesh: meshResource, materials: [material])
modelEntity.name = "modelEntity_targetShape"
// anchorEntity
let anchorEntity = AnchorEntity (plane: .any)
anchorEntity.addChild (modelEntity)
anchorEntity.name = "anchorEntity_targetShape"
return anchorEntity
// ...
// Moving
targetShape.move (to: transform, relativeTo: nil, duration: 0.4, timingFunction: .easeOut)
// ...
```
This works, but there is a noticeable lag in texture rendering when moving, moreover, when changing the ARCamera.TrackingState status other than **.normal**, unpredictable texture behavior is observed.

Thanks!

### PR DESCRIPTION
**Pull request by Reality-Dev**
Now you can upload your own image textures and use them for the diffuse of your focus entity, instead of just the segment edges or a plane mesh. 
      try .texture(.load(named: "Add"))
^^put your image in Assets.xcassets, and then place its (exact) name in that string above.
Transparency will be respected (e.g. if you upload a png texture that looks like a circle, the focus entity will look like a circle).
---
     static var defaultPlane: MeshResource {}
^^this plane was showing up far too large, so I commented it out.

### CODE REVIEW

File: FocusEntity-Example/FocusEntity-Example/FocusARView.swift [hunk 1]
```diff
@@ -17,8 +17,18 @@ class FocusARView: ARView {
   required init(frame frameRect: CGRect) {
     super.init(frame: frameRect)
     self.setupConfig()
-    self.focusEntity = FocusEntity(on: self, focus: .classic)
-//    self.focusEntity = FocusEntity(on: self, style: .colored(onColor: .red, offColor: .blue, nonTrackingColor: .orange))
+    //self.focusEntity = FocusEntity(on: self, focus: .classic)
```
Comment (maxxfrazer) [inline FocusEntity-Example/FocusEntity-Example/FocusARView.swift]: Let's leave the example project showing the classic version, with the texture version commented

File: FocusEntity-Example/FocusEntity-Example/FocusARView.swift [hunk 2]
```diff
@@ -17,8 +17,18 @@ class FocusARView: ARView {
...
+    //self.focusEntity = FocusEntity(on: self, focus: .plane)
+    do{
+        let onColor : MaterialColorParameter = try .texture(.load(named: "Add"))
+        let offColor : MaterialColorParameter = try .texture(.load(named: "Open"))
```
Comment (maxxfrazer) [inline FocusEntity-Example/FocusEntity-Example/FocusARView.swift]: bit nit-picky, but could you standardise the colons here removing that first space? (`onColor: Materi...`)

File: Sources/FocusEntity/FocusEntity+Colored.swift [hunk 3]
```diff
@@ -8,15 +8,16 @@
 
 #if canImport(ARKit)
 import RealityKit
+import UIKit
 
 /// An extension of FocusEntity holding the methods for the "colored" style.
 public extension FocusEntity {
-
+
```
Comment (maxxfrazer) [inline Sources/FocusEntity/FocusEntity+Colored.swift]: Please delete these trailing spaces

File: Sources/FocusEntity/FocusEntity+Colored.swift [hunk 4]
```diff
@@ -25,9 +26,12 @@ public extension FocusEntity {
...
         self.fillPlane?.model?.materials = [SimpleMaterial()]
     }
-    self.fillPlane?.model?.materials[0] = SimpleMaterial(
-      color: endColor, isMetallic: false
-    )
+    //Necessary for transparency.
+    var modelMaterial = UnlitMaterial(color: .clear)
+    modelMaterial.baseColor = endColor
+    //Necessary for transparency.
+    modelMaterial.tintColor = UIColor.white.withAlphaComponent(0.99)
```
Comment (maxxfrazer) [inline Sources/FocusEntity/FocusEntity+Colored.swift]: Use `Material.Color.white...` instead, then remove UIKit import

File: Sources/FocusEntity/FocusEntityComponent.swift [hunk 5]
```diff
@@ -51,25 +54,28 @@ public struct FocusEntityComponent: Component {
...
   public static let classic = FocusEntityComponent(style: .classic(color: #colorLiteral(red: 1, green: 0.8, blue: 0, alpha: 1)))
   public static let plane = FocusEntityComponent(
     style: .colored(
-      onColor: .green,
-      offColor: .orange,
-      nonTrackingColor: Material.Color.red.withAlphaComponent(0.2),
-      mesh: FocusEntityComponent.defaultPlane
+        onColor: MaterialColorParameter.color(.green),
+      offColor: MaterialColorParameter.color(.orange),
+        nonTrackingColor: MaterialColorParameter.color(UIColor.red.withAlphaComponent(0.2)),
```
Comment (maxxfrazer) [inline Sources/FocusEntity/FocusEntityComponent.swift]: Same here for removing UIKit & `Material.Color.red.with...`

File: Sources/FocusEntity/FocusEntityComponent.swift [hunk 6]
```diff
@@ -51,25 +54,28 @@ public struct FocusEntityComponent: Component {
...
+        mesh: MeshResource.generatePlane(width: 0.1, depth: 0.1)
```
Comment (maxxfrazer) [inline Sources/FocusEntity/FocusEntityComponent.swift]: Could you fix the spacing here? (mix of 2 and 4 spaces, rest of the repo uses 2 spaces)

File: Sources/FocusEntity/FocusEntityComponent.swift [hunk 7]
```diff
@@ -51,25 +54,28 @@ public struct FocusEntityComponent: Component {
...
-    return MeshResource.generatePlane(width: length, depth: length)
-  }
+//  static var defaultPlane: MeshResource {
+//    let thickness = 0.018
+//    let correctionFactor = thickness / 2 // correction to align lines perfectly
+//    let length = Float(1.0 - thickness * 2 + correctionFactor)
+//
+//    return MeshResource.generatePlane(width: length, depth: length)
+//  }
+
```
Comment (maxxfrazer) [inline Sources/FocusEntity/FocusEntityComponent.swift]: Go ahead and delete this, I think it was used in a previous version, d'oh

**Review by maxxfrazer — CHANGES_REQUESTED** [review]
Thanks for this PR! Great to have this previously overlooked feature added; I like the use of MaterialColorParameter, really makes it simple to choose between a plain colour and a texture.
Feel free to add a section at the bottom of the Usage section of the README if you have chance, otherwise I can add it with the 1.3.0 release.

A few changes but they're all small ones!
I just realised the SwiftLint Github Action isn't working right on this repo, I'll fix that and adjust the other problems after this is merged.

**Review by maxxfrazer — APPROVED** [review]

**Comment by Reality-Dev** [comment]
I just saw these extra comments. Thanks for getting it fixed up and ready to go so quickly! I have more updates on the way coming soon (which will probably require substantially more revisions). Great work!

**Pull request merged**